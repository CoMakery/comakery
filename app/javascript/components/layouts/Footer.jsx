import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import Icon from './../styleguide/Icon'

class Footer extends React.Component {
  render() {
    const {className, isLoggedIn, ...other} = this.props

    const classnames = classNames(
      'footer',
      className
    )

    return (
      <React.Fragment>
        <div className={classnames} {...other}>
          <div className="footer--logo">
            <Icon name="Logo-Footer.svg" width="170px" />
          </div>

          <div className="footer--content">
            <div className="footer--content--wrapper">
              <div className="footer--content--text">
                CoMakery helps you gather a tribe to achieve big missions.
                <br />
                To achieve your big mission you need to bring people together around a common vision, let them know what they will get by contributing, and organize the work. The CoMakery platform helps you do this with missions, projects, task workflows, tokens & payments.
              </div>

              <div className="footer--content--social">
                <a href="https://twitter.com/comakery" target="_blank"><img src={require(`src/images/nucleo-social-icons/social-1_logo-twitter.svg`)} /></a>
                <a href="https://www.facebook.com/comakery" target="_blank"><img src={require(`src/images/nucleo-social-icons/social-1_logo-facebook.svg`)} /></a>
                <a href="https://www.linkedin.com/company/comakery/about" target="_blank"><img src={require(`src/images/nucleo-social-icons/social-1_logo-linkedin.svg`)} /></a>
              </div>
            </div>

            <div className="footer--content--nav">
              <div className="footer--content--nav--about">
                <div className="footer--content--nav--about--header">
                  About CoMakery
                </div>
                <a href="/">
                  Home
                </a>
                <a className="intercom" href="">
                  Contact Us
                </a>
                <a href="https://ledger.comakery.com" target="_blank">
                  Blog
                </a>
              </div>

              { !isLoggedIn &&
                <div className="footer--content--nav--join">
                  <div className="footer--content--nav--join--header">
                    Join
                  </div>
                  <a href="/accounts/new">
                    Contributors
                  </a>
                  <a href="/accounts/new">
                    Foundations
                  </a>
                </div>
              }

              <div className="footer--content--nav--legal">
                <div className="footer--content--nav--legal--header">
                  Legal
                </div>
                <a href="/user-agreement">
                  User Agreement
                </a>
                <a href="/prohibited-use">
                  Prohibited Use
                </a>
                <a href="/e-sign-disclosure">
                  E-Sign Disclosure
                </a>
                <a href="/privacy-policy">
                  Privacy Policy
                </a>
              </div>
            </div>
          </div>

          <div className="footer--copyright">
            Â© {(new Date()).getFullYear()} CoMakery, Inc. All rights reserved.
          </div>
        </div>
      </React.Fragment>
    )
  }
}
Footer.propTypes = {
  isLoggedIn: PropTypes.bool
}
Footer.defaultProps = {
  isLoggedIn: false
}
export default Footer
