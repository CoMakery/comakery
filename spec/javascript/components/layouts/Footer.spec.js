import React from 'react'
import { shallow } from 'enzyme'
import Footer from 'components/layouts/Footer'

describe('Footer', () => {
  it('renders correctly without props', () => {
    const wrapper = shallow(<Footer />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.exists('.footer')).toBe(true)
    expect(wrapper.exists('.footer--logo')).toBe(true)
    expect(wrapper.exists('.footer--content')).toBe(true)
    expect(wrapper.exists('.footer--content--text')).toBe(true)
    expect(wrapper.exists('.footer--content--nav')).toBe(true)
    expect(wrapper.exists('.footer--copyright')).toBe(true)
  })

  it('renders correctly with custom className', () => {
    const wrapper = shallow(<Footer className="__test" />)

    expect(wrapper.exists('.footer.__test')).toBe(true)
  })
})
