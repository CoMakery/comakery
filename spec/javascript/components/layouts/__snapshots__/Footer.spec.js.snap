// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly without props 1`] = `
<Fragment>
  <div
    className="footer"
  >
    <div
      className="footer--logo"
    >
      White Logo Placeholder
    </div>
    <div
      className="footer--content"
    >
      <div
        className="footer--content--text"
      >
        CoMakery accelerates blockchain protocol adoption with scalable teams of talented makers. We build communities of high-level, distributed engineers, designers, and community managers, and match them with unique blockchain projects.
      </div>
      <div
        className="footer--content--nav"
      >
        <div
          className="footer--content--nav--about"
        >
          <div
            className="footer--content--nav--about--header"
          >
            About CoMakery
          </div>
          <a
            href="/"
          >
            Home
          </a>
          <a
            href="mailto:support@comakery.com"
          >
            Contact Us
          </a>
        </div>
        <div
          className="footer--content--nav--join"
        >
          <div
            className="footer--content--nav--join--header"
          >
            Join
          </div>
          <a
            href="/accounts/new"
          >
            Contributors
          </a>
          <a
            href="/accounts/new"
          >
            Foundations
          </a>
        </div>
        <div
          className="footer--content--nav--legal"
        >
          <div
            className="footer--content--nav--legal--header"
          >
            Legal
          </div>
          <a
            href="/user-agreement"
          >
            User Agreement
          </a>
          <a
            href="/prohibited-use"
          >
            Prohibited Use
          </a>
          <a
            href="/e-sign-disclosure"
          >
            E-Sign Disclosure
          </a>
          <a
            href="/privacy-policy"
          >
            Privacy Policy
          </a>
        </div>
      </div>
    </div>
    <div
      className="footer--copyright"
    >
      Â© 
      2018
       CoMakery, Inc. All rights reserved.
    </div>
  </div>
</Fragment>
`;
